modules/programs/calcurse.nix

```
@@ -0,0 +1,191 @@
+    2. dd/mm/yyyy
+    3. yyyy/mm/dd
+    4. yyyy-mm-dd
+        '';
+      };
+
+      notifyDate = mkOption {
+        type = types.str;
+        default = "%a %F";
+        description = "With this option, you can specify the format to be used to display the current date inside the notification bar. You can see all of the possible formats by typing man 3 strftime inside a terminal.";
+      };
+
+      notifyTime = mkOption {
+        type = types.str;
+        default = "%T";
+        description = "With this option, you can specify the format to be used to display the current time inside the notification bar. You can see all of the possible formats by typing man 3 strftime inside a terminal.";
+      };
+    };
+
+    notification = {
+      warning= mkOption {
+        type = types.int;
+        default = 300;
+        description = "When there is an appointment which is flagged as important within the next notification.warning seconds, the display of that appointment inside the notify-bar starts to blink. Moreover, the command defined by the notification.command option will be launched. That way, the user is warned and knows there will be soon an upcoming appointment.";
+      };
+
+      command = mkOption {
+        type = types.str;
+        default = "printf '\\a'";
+        description = "This option indicates which command is to be launched when there is an upcoming appointment flagged as important. This command will be passed to the user’s shell which will interpret it. To know what shell must be used, the content of the $SHELL environment variable is used. If this variable is not set, /bin/sh is used instead.";
+      };
+
+      notifyAll = mkOption {
+        type = types.str;
+        default = "flagged-only";
+        description = "If set to flagged-only, you are only notified of flagged items. If set to unflagged-only, you are only notified of unflagged items. If set to all, notifications are always sent, independent of whether an item is flagged or not. For historical reasons, this option also accepts boolean values where yes equals flagged-only and no equals unflagged-only.";
+      };
+    };
+
+    daemon = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "If set to yes, daemon mode will be enabled, meaning calcurse will run into background when the user’s interface is exited. This will allow the notifications to be launched even when the interface is not running.";
+      };
+
+      log = mkOption {
+        type = types.bool;
+        default = false;
+        description = "If set to yes, calcurse daemon activity will be logged.";
+      };
+    };
+
+
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.calcurse ];
+
+    home.file.".calcurse/conf".text = concatStringsSep "\n" (
+      map (g: "general." + g) (mapAttrsToList formatLine cfg.general) ++
+      map (a: "appearance." + a) (mapAttrsToList formatLine cfg.appearance) ++
```

It's preferred to place the `++` in the beginning of the line. Like
```suggestion
      ++ map (a: "appearance." + a) (mapAttrsToList formatLine cfg.appearance)
```